src/charm.py

```diff
@@ -35,6 +36,11 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.gunicorn_pebble_ready, self._on_gunicorn_pebble_ready)
 
+        self.mongodb = DatabaseRequires(self, relation_name="mongodb_client", database_name=self.app.name)
+        self.framework.observe(self.mongodb.on.database_created, self._on_mongodb_client_relation_changed)
+        self.framework.observe(self.on.mongodb_client_relation_changed, self._on_mongodb_client_relation_changed)
+        self.framework.observe(self.on.mongodb_client_relation_joined, self._on_mongodb_client_relation_changed)
```

Comment:
Will these be triggered if new units are added to the client application?

Reply:
The `endpoints_changed` event must be emitted on database horizontal scaling (adding/removing units).
But the mongodb-k8s-operator charm is not doing that right now, and it should be fixed on the mongodb-k8s-operator side.
Feel free to issue a ticket. It is an issue but not critical because the MongoDB client driver is receiving replicaset members as a part of connection establishment (it means even if we have the wrong list of members in the config, the application will connect to the proper list of members anyway).

Reply:
I'm talking about when the related application is scaled, not when mongodb is scaled here. Will these `database_created` and `endpoints_changed` events be triggered in that case? If not I think we'd still need to keep the `relation_joined` one at least.

Reply:
Have confirmed that the `database_created` event is triggered when a new unit is added to the client application. Thanks.